saml-core/src/main/java/org/keycloak/saml/processing/core/saml/v2/util/AssertionUtil.java

```diff
@@ -272,24 +272,21 @@ public static void createSAML11TimedConditions(SAML11AssertionType assertion, lo
      * @param assertionElement
      * @param publicKey the {@link PublicKey}
      *
-     * @return
+     * @return true if signature is valid
      */
     public static boolean isSignatureValid(Element assertionElement, PublicKey publicKey) {
-        try {
-            Document doc = DocumentUtil.createDocument();
-            Node n = doc.importNode(assertionElement, true);
-            doc.appendChild(n);
-
-            return new SAML2Signature().validate(doc, new HardcodedKeyLocator(publicKey));
-        } catch (Exception e) {
-            logger.signatureAssertionValidationError(e);
-        }
-        return false;
+        return isSignatureValid(assertionElement, new HardcodedKeyLocator(publicKey));
     }
 
     /**
-     * Given an assertion element, validate the signature.
+     * Given an assertion element, validate the signature
+     *
+     * @param assertionElement
+     * @param keyLocator the {@link KeyLocator}
+     *
+     * @return true if signature is valid
      */
+    
     public static boolean isSignatureValid(Element assertionElement, KeyLocator keyLocator) {
         try {
             Document doc = DocumentUtil.createDocument();
```

Comment:
Fix to the previous PR that introduced this method - this method should use XMLSignatureUtil.validateSingleNode() (make it public beforehand) instead of reimplementing the logic here.